### PR TITLE
added documentation on how to reduce overhead

### PR DESCRIPTION
**Pull request by beniwohli**
also, added documentation for the sample_rate config, and reduced
the flush_interval setting to 10s

### CODE REVIEW

File: docs/tuning.asciidoc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Using an APM solution comes with certain trade-offs, and the Python agent for Elastic APM is no different.
+Instrumenting your code, measuring timings, recording context data etc. all need resources: 
+
+ * CPU time
+ * memory
+ * bandwidth use
+ * Elasticsearch storage
+
+We invested and continue to invest a lot of effort to keep the overhead of using Elastic APM as low as possible.
+But because every deployment is different, we added some knobs you can turn to adapt it to your specific needs.
```
Comment (roncohen) [inline docs/tuning.asciidoc]: we have added

File: docs/tuning.asciidoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+[float]
+[[tuning-sample-rate]]
+=== Transaction Sample Rate
+
+The most straight forward way to reduce the overhead of the agent is to tell the agent to do less.
+If you set the <<config-transaction-sample-rate,`transaction_sample_rate`>> to a value bellow `1.0`,
```
Comment (roncohen) [inline docs/tuning.asciidoc]: below

File: docs/tuning.asciidoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+the agent will randomly sample only a subset of transactions.
+If a transaction is not sampled, the agent has to do a lot less work,
+as we only record the overall transaction time and the result for unsampled transactions.
```
Comment (roncohen) [inline docs/tuning.asciidoc]: and the transaction name

File: docs/tuning.asciidoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Reducing the sample rate to a fraction of all transactions can make a huge difference in all four of the mentioned resource types.
+
+[float]
+[[tuning-queue]]
+=== Transaction Queue
+
+To reduce the load on the APM server, the agent does not send every transaction up as it happens.
+Instead, it queues them up, and flushes the queue periodically, or when it reaches a maximum size, using a background thread.
+
+While this reduces the load on the APM server (and to a certain extent on the agent),
```
Comment (roncohen) [inline docs/tuning.asciidoc]: APM Server (capital s)

File: docs/tuning.asciidoc (hunk 1 above)
Comment (roncohen) [reply docs/tuning.asciidoc]: or "there are"

Comment (beniwohli) [reply docs/tuning.asciidoc]: hmm yes, "there are" seems better

**Pull request closed without merging**